databuilder/models/table_metadata.py

```diff
@@ -241,6 +241,7 @@ def __init__(self,
                  is_view: bool = False,
                  tags: Union[List, str] = None,
                  description_source: Union[str, None] = None,
+                 badges: Union[List[str], None] = None,
```

Comment:
hey Diksha, with https://github.com/amundsen-io/amundsendatabuilder/blob/master/databuilder/models/badge.py#L22, any additional do we still need to add badge into table model? cc @allisonsuarez

Reply:
Yes, we have to first add Badge and then create BadgeMetadata from TableMetadata while creating nodes and relations. This is similar to column badge implementation. Let me know if there is another way to doing it.

Reply:
I don't think you need to put the badge creation inside table as you could build badge<->table relationship just with the badge model, or i miss anything?

Reply:
so this field only contains the badge strings coming in, but then below we can see that BadgeMetadata is called for node and relation creation.

Reply:
This follows a similar pattern to the way col level badges are added when creating the table metadata https://github.com/amundsen-io/amundsendatabuilder/blob/86229c05ee926b5bf815c344d2942b9bcb1c359b/databuilder/models/table_metadata.py#L396

Reply:
I see, then questions:
1. would it be a bit duplicate now given you will have two ways to create badges?
2. should it be a list of Badge object instead of string?

Reply:
I see what you mean, in that case we could just create the badge when extracting the table badge information.

Reply:
yeah, otherwise everytime we need to add the same code to table, dashboard, etc which needs badge integration.

Reply:
^^ @dikshathakur3119 let me know if it makes sense. I think for table deprecation badge, you could build an extractor that extract table deprecation info and use the badge object to create the relationship. This is more flexible than putting the code in table model.

Reply:
Thanks Tao, it makes sense to me. I will add a different extractor for adding BadgeMetadata.

Reply:
So I will close this PR since I would not need any change to TableMetadata?